app/driver.go

```diff
@@ -109,12 +109,35 @@ func deployDriver(c *cli.Context) error {
 		return errors.Wrap(err, "unable to get k8s client")
 	}
 
+	driverDetected := false
 	driver := c.String(FlagDriver)
 	if driver == "" {
+		driverDetected = true
 		driver, err = chooseDriver(kubeClient)
+		logrus.Debugf("Driver %s will be used after automatic detection", driver)
 		if err != nil {
 			return err
 		}
+	} else {
+		logrus.Debugf("Explicitly choose the %s driver", driver)
+	}
+
+	if driver == FlagDriverCSI {
+		csiChecked, err := csi.CheckMountPropagationWithPodSpec(kubeClient, managerImage, os.Getenv(types.EnvPodNamespace))
```

Comment:
The function can just return an error, including when it's detected mount propagation was not enabled, since the case that `false, nil` is handled in the same way as `nil, err`.